include/kframework/gas.md

```diff
@@ -0,0 +1,50 @@
+KEVM Gas
+========
+
+Here is a representation of `Gas`. This sort is used by `<gas>`, `<callGas>`, and `<gasUsed>` cells of the EVM configuration. As is in this file, `Gas` is a super-sort of `Int` with no differing behaviour. However the distinction at this level allows for an extension of infinite gas later, see `infinite-gas.md`.
+
+```k
+module GAS-SYNTAX
+    imports INT-SYNTAX
+    
+    syntax Gas ::= Int
+    syntax Int ::= "gas2Int" "(" Gas ")" [function, total]
```

Comment:
```suggestion
    syntax Int ::= "gas2Int" "(" Gas ")" [function]
```

I don't think this function is total, is it?

I guess for `gas2Int(#gas(X)) => X` we kinda make it total, but does that really make sense? That means that `#gas` doesn't really behave like a point, does it?

Reply:
I agree, I think this is the weakest part of the PR. `gas2Int` only exists to process the `GAS` opcode by placing the `<gas>` cell on the wordstack, so a `#gas(G)` needs to be stripped to `G` so the sort doesn't conflict. But I think this as a solution is flawed, as the information that `G` came from a `#gas()` is lost.